rs/index/src/ivf/writer.rs

```diff
@@ -280,7 +285,8 @@ mod tests {
         // Create an IvfWriter instance
         let num_features = 10;
         let quantizer = NoQuantizer::new(num_features);
-        let ivf_writer = IvfWriter::new(base_directory.clone(), quantizer);
+        let ivf_writer: IvfWriter<NoQuantizer, L2DistanceCalculator> =
```

Comment:
Genuine question: could this be just `IvfWriter<_, L2DistanceCalculator>`? I see that we do that for `write_quantizer_and_build_ivf_index::<_, L2DistanceCalculator, _>` down below, so trying to see if we can be more consistent here.